In [28]:
import numpy as np
import pandas as pd
from scipy.stats import describe
from scipy import stats

In [29]:
df=pd.read_csv(r"C:\Users\surya\Downloads\LabTAT.csv")
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [30]:
df.rename(columns={'Laboratory 1':'Lab1' ,'Laboratory 2':'Lab2','Laboratory 3':'Lab3','Laboratory 4':'Lab4'},inplace=True)
D=describe(df)
df.head()

Lab1    Lab2    Lab3    Lab4
0  185.35  165.53  176.70  166.13
1  170.49  185.91  198.45  160.79
2  192.77  194.92  201.23  185.18
3  177.33  183.00  199.61  176.42
4  193.41  169.57  204.63  152.60

In [31]:
var1=df['Lab1'].var()
var2=df['Lab2'].var()
var3=df['Lab3'].var()
var4=df['Lab4'].var()

In [32]:
print("Inferences from the data")
for key,value in D._asdict().items():
    print(f"{key}:{value}")
Variance=[var1,var2,var3,var4]
print("Variance:",Variance)

Inferences from the data
nobs:120
minmax:(array([138.3 , 140.55, 159.69, 124.06]), array([216.39, 217.86, 238.7 , 205.18]))
mean:[178.36158333 178.90291667 199.91325    163.68275   ]
variance:[173.54356806 223.71525109 273.539607   227.55963187]
skewness:[-0.17486957  0.04014906  0.13190811 -0.10272477]
kurtosis:[ 0.67167075 -0.19500961 -0.42915427  0.31065064]
Variance: [173.54356806022412, 223.71525108543426, 273.53960699579824, 227.55963186974788]


Given
confidence level = 95%

α = 0.05 (1-0.95)

H0: All the Laboratories are normal

H1: All the Laboratories are not normal

## Normality Test 

In [33]:
print(stats.shapiro(df['Lab1']))
print(stats.shapiro(df['Lab2']))
print(stats.shapiro(df['Lab3']))
print(stats.shapiro(df['Lab4']))

ShapiroResult(statistic=0.990183413028717, pvalue=0.5507769584655762)
ShapiroResult(statistic=0.9936320185661316, pvalue=0.8637352585792542)
ShapiroResult(statistic=0.988633930683136, pvalue=0.42045730352401733)
ShapiroResult(statistic=0.9913760423660278, pvalue=0.6619619131088257)


# Comparing the P-values

P-Value of Lab1 = 0.55>α

P-Value of Lab2 = 0.86>α

P-value of Lab3 = 0.42>α

P-Value of Lab4 = 0.66>α

So, we can conclude that all the p-values are greater than the significance level.


### H0 is accepted. All the Laboratories are Normal

Now, We will check the Variance of laboratories...

## Variance Test

H0: There is no siginficance between the variances

H1: There is a siginficance between the variances

In [34]:
stats.levene(df["Lab1"],df["Lab2"],df["Lab3"],df["Lab4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

Hence, relative pvalue of all variances > α
i.e, 0.0516>0.05

### H0 is accepted, There is no significance difference between the variances of laboratories

Now, we will calculate the ANOVA Test.

H0 = The Mean TAT for all the laboratories is same

H1 = The Mean TAT for all the laboratories is not same

In [35]:
stats.f_oneway(df["Lab1"], df["Lab2"], df["Lab3"], df["Lab4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Here, The p-value is less than α

2e-57 < α


### Therefore, H1 is accepted, There is a significance difference between the MEAN TAT.